### PR TITLE
Feat: attachments automation

### LINKED ISSUE #4054: Add automation action send attachment
Add automation action send attachment

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Send an attachment that is attached to the theutomation rule in the message when conditions are met. 

Fixes #4054 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Added test cases for the same.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/models/automation_rule.rb [hunk 1]
```diff
@@ -19,29 +19,34 @@
 #
 class AutomationRule < ApplicationRecord
   belongs_to :account
+  has_one_attached :file
```
Comment (sojan-official) [inline app/models/automation_rule.rb:22]: has_many_attached ? 
ref: https://edgeguides.rubyonrails.org/active_storage_overview.html#has-many-attached

File: app/models/automation_rule.rb [hunk 2]
```diff
@@ -19,29 +19,34 @@
...
 
   validate :json_conditions_format
   validate :json_actions_format
   validates :account_id, presence: true
 
   scope :active, -> { where(active: true) }
 
-  CONDITIONS_ATTRS = %w[country_code status browser_language assignee_id team_id referer].freeze
-  ACTIONS_ATTRS = %w[send_message add_label send_email_to_team assign_team assign_best_agents].freeze
+  CONDITIONS_ATTRS = %w[country_code status message_type browser_language assignee_id team_id referer is_priority_customer company].freeze
```
Comment (sojan-official) [inline app/models/automation_rule.rb:30]: do we have an attribute `is_priority_customer `?

File: app/services/automation_rules/action_service.rb [hunk 3]
```diff
@@ -21,6 +21,14 @@ def perform
 
   private
 
+  def send_attachments(_file_params)
+    return if @rule.event_name == 'message_created'
+
+    params = { content: nil, private: false, attachments: [@rule.file_blob] }
+    mb = Messages::MessageBuilder.new(@administrator, @conversation, params)
```
Comment (sojan-official) [inline app/services/automation_rules/action_service.rb:28]: where do get the value of `@administrator` from?

**Review by sojan-official — COMMENTED** [review]
added some review comments. back to you

File: app/services/automation_rules/action_service.rb [hunk 4]
```diff
@@ -21,6 +21,14 @@ def perform
 
   private
 
+  def send_attachments(_file_params)
+    return if @rule.event_name == 'message_created'
+
+    params = { content: nil, private: false, attachments: [@rule.file_blob] }
```
[the remark this answers is not in the document]
Comment (tejaswinichile) [reply app/services/automation_rules/action_service.rb:27]: No, not yet. Considering we are sending one file after the automation conditions meet.

Let me remove the plural.

File: app/models/automation_rule.rb (hunk 2 above)
Comment (tejaswinichile) [reply app/models/automation_rule.rb:30]: It's actually custom_attribute. Need to handle it differently.

File: app/services/automation_rules/action_service.rb (hunk 3 above)
Comment (tejaswinichile) [reply app/services/automation_rules/action_service.rb:28]: From here: 
https://github.com/chatwoot/chatwoot/pull/4266/files#diff-f127d9ac5d9df6d9a8903e5929efde7a571cec4c374ff2f13dc61ce68ca8b2c2R91

**Comment by tejaswinichile** [comment]
@sojan-official I have fixed the PR comments, can you check this?

File: app/services/automation_rules/action_service.rb (hunk 3 above)
Comment (sojan-official) [reply app/services/automation_rules/action_service.rb:29]: *https://github.com/chatwoot/chatwoot/blob/971755b84570e8c23f83193dd98fec0f93482356/app/services/automation_rules/action_service.rb#L82

**Comment by sojan-official** [comment]
@tejaswinichile  Just noted that we are using `@account.administrators.first` as the message sender for the attachment message. Since this behaviour is already present for send message automation, let's take it up as a separate discussion. 

Otherwise, the PR looks good to me.

**Comment by tejaswinichile** [comment]
@sojan-official This is ready to review, made the changes for message_sender nil.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**